src/ui/public/saved_objects/saved_objects_client_provider.js

```diff
@@ -1,7 +1,21 @@
 import chrome from 'ui/chrome';
+import { ErrorAutoCreateIndexProvider } from 'ui/error_auto_create_index';
 
 import { SavedObjectsClient } from './saved_objects_client';
 
-export function SavedObjectsClientProvider($http, $q) {
-  return new SavedObjectsClient($http, chrome.getBasePath(), $q);
+export function SavedObjectsClientProvider($http, $q, Private) {
+  const errorAutoCreateIndex = Private(ErrorAutoCreateIndexProvider);
+
+  return new SavedObjectsClient({
+    $http,
+    basePath: chrome.getBasePath(),
```

Comment:
__nit:__ you can probably omit `basePath` property since it's the same as default one.